Problem4/src/problem4/Problem4.java

```diff
@@ -15,7 +22,105 @@
+                 lines++;
+                reval.append(contentLine);
+                reval.append(" ");
+            }
+        } catch (IOException ioe) {
+            ioe.printStackTrace(System.out);
+        } finally {
+            try {
+                if (readFile != null) {
+                    readFile.close();
+                }
+            } catch (IOException ioe) {
+                System.out.println("Error in closing the BufferedReader");
+                ioe.printStackTrace(System.out);
+            }
+        }
+        return new returnValues(reval.toString(), lines);
+    }
+    public static int wordCount(String input) {
+        String[] size = input.split("\\s+");
+        return size.length;
+    }
+
+    public static String longestWord(String input) {
+        String longestWord = null;
+        int length = 0;
+        String[] words = input.split(" ");
+        for (String word : words) {
+            if (length < word.length()) {
+                length = word.length();
+                longestWord = word;
+            }
+        }
+        return longestWord;
+    }
+    public static int longestWordLength(String input) {
+        return input.length();
+    }
+    
+    public static double averageWordLength(String input) {
+        double sum = 0;
+        String[] words = input.split("\\s+");  
+        for (int i = 0; i < words.length; i++) {
+            words[i] = words[i].trim();
+            sum += words[i].length();
+        }
+        return  sum / words.length;
+    }
+    public static void outputDataText(int wordCount, int lineCount, String longestWord, int longestWordLength, double averageWordLength, String fileName) {
+        try {
+            File output = new File("myTextData.txt");
+            if (!output.exists()) {
+                output.createNewFile();
+            }
+            FileWriter outputFW = new FileWriter(output, false);
+            BufferedWriter outputBW = new BufferedWriter(outputFW);
+            PrintWriter outputPW = new PrintWriter(outputBW);
+
+            outputPW.println("The file " + fileName + " was successfully read.\n");
+            outputPW.printf("%-21s%-5d", "Number of words: ", wordCount);
+            outputPW.println();
+            outputPW.printf("%-21s%-5d", "Number of lines: ", lineCount);
+            outputPW.println();
+            outputPW.printf("%-21s%-5s", "Longest word: ", longestWord);
+            outputPW.println();
+            outputPW.printf("%-21s%-5s", "Longest word length: ", longestWordLength);
+            outputPW.println();
+            outputPW.printf("%-21s%-5.1f", "Average word length: ", averageWordLength);
+       
+            outputPW.close();
+            System.out.println("Data successfully written to myTextData.txt");
+        } catch (IOException ioe) {
+            System.out.println("Exception occurred:");
+            ioe.printStackTrace(System.out);
+        }
     }
     
 }
+ class returnValues {
```

Comment:
As best practice, you should put your _returnValues_ class in its own file (and the name should be _ReturnValues_).